apps/dgst.c

```diff
@@ -417,11 +427,13 @@ int dgst_main(int argc, char **argv)
                 ret++;
                 continue;
             } else {
-                r = do_fp(out, buf, inp, separator, out_bin, sigkey, sigbuf,
-                          siglen, sig_name, md_name, argv[i]);
+                r = do_fp(out, buf, inp, separator, out_bin, xoflen,
+                          sigkey, sigbuf, siglen, sig_name, md_name, argv[i]);
             }
             if (r)
                 ret = r;
+            /* EVP_DigestFinalXOF resets the MD */
+            (void)BIO_set_md(bmd, md);
```

Comment:
This should not be necessary IMO. The following BIO_reset call should handle it fine.

Reply:
I agree; this is because the use of `EVP_DigestFinalXOF` in `do_fp` breaks the invariant where `md` is always set in the underlying ctx, when the BIO is marked as initialized, with `init = 1`.

Reply:
Ha, this is actually a bug/API regression in 3.0.

Reply:
See #13402
So please remove that call and we should rebase this upon the #13402 once it is merged.

Reply:
@t8m rebased against the master now that #13402 has been merged, though I still like to hear any opinions on https://github.com/openssl/openssl/pull/13245#issuecomment-729080953.